src/Tribe/Rewrite.php

```diff
@@ -0,0 +1,166 @@
+<?php
+// Don't load directly
+defined( 'WPINC' ) or die;
+
+/**
+ * Rewrite Configuration Class
+ * Permalinks magic Happens over here!
+ */
+class Tribe__Tickets__Rewrite extends Tribe__Rewrite {
+
+	/**
+	 * Tribe__Tickets__Rewrite constructor.
+	 *
+	 * @param WP_Rewrite|null $wp_rewrite
+	 */
+	public function __construct( WP_Rewrite $wp_rewrite = null ) {
+		$this->rewrite = $wp_rewrite;
+	}
+
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 * @return Tribe__Tickets__Rewrite
+	 */
+	public static function instance( $wp_rewrite = null ) {
```

Comment:
Do not use singletons.
Again I know existing examples in code are confusing but we're trying to keep classes unaware of their usage context; register the class with a slug with `tribe_singleton()` and call it with `tribe($slug)` where you need to use the service locator pattern and cannot use dependency injection.
E.g.:
```php
// here we say that `tickets.rewrite` means the same instance of Tribe__Tickets__Rewrite
tribe_singleton( 'tickets.rewrite', 'Tribe__Tickets__Rewrite' );

// ...in another piece of code we use the service locator
$rewrite = tribe( 'tickets.rewrite' );
$rewrite->do_something();
```

Reply:
Update the code using the `instance` method of this class to use `tribe( 'slug' )`